spec.md

```diff
@@ -549,6 +549,19 @@ Analog<32> ; 32-bit analog type
 Analog     ; analog type with inferred width
 ```
 
+### Enumeration Types
+
+Enumerations are structural disjoint union types.  An enumeration has a number
+of variants, each with a type.  The different variants are specified with tags.
+A variant may optionally omit the data type, in which case it is implicitly
+defined to be `UInt<0>`. The variant types of an enumeration must all be
+passive and cannot contain analog types.
+
+``` firrtl
+{|a, b, c|}
+{|some: UInt<8>, none|}
+```
```

Comment:
Some comments or commentary that `a`, `b`, and `c` are "tags" and `UInt<8>` is a "type" would help make it obvious what is going on.

Reply:
Reworded this section, please take a look.